build/url.go

```diff
@@ -44,8 +48,8 @@ func createTempDockerfileFromURL(ctx context.Context, d *driver.DriverHandle, ur
 		if err != nil {
 			return nil, err
 		}
-		if proto.Size(stat) > 512*1024 {
-			return nil, errors.Errorf("Dockerfile %s bigger than allowed max size", url)
+		if stat.Size > maxDockerfileSizeBytes {
+			return nil, errors.Errorf("Dockerfile %s bigger than allowed max size (%s)", url, maxDockerfileSizeHumanReadable)
```

Comment:
For string value use `maxDockerfileSizeBytes * units.B` https://pkg.go.dev/github.com/tonistiigi/units or `units.HumanSize` https://pkg.go.dev/github.com/docker/go-units#HumanSize . Both are already vendored.